include/crow/utility.h

```diff
@@ -627,125 +625,78 @@ namespace crow
 
         inline static void sanitize_filename(std::string& data, char replacement = '_')
         {
-            unsigned char i = 0, length_limit;
-
-            length_limit = data.length() < 255 ? data.length() : 255;
-            data = data.substr(0, length_limit);
+            if (data.length() > 255)
+                data.resize(255);
 
-            for (; i < length_limit; i++)
+            static const auto toUpper = [](char c) {
+                return ((c >= 'a') && (c <= 'z')) ? (c - ('a' - 'A')) : c;
+            };
+            // Check for special device names. The Windows behavior is really odd here, it will consider both AUX and AUX.txt
+            // a special device. Thus we search for the string (case-insensitive), and then check if the string ends or if
+            // is has a dangerous follow up character (.:\/)
+            auto sanitizeSpecialFile = [](std::string& source, unsigned ofs, const char* pattern, bool includeNumber, char replacement) {
+                unsigned i = ofs, len = source.length();
```

Comment:
Would it be better if `i` and `len` were `unsigned char` (or `uint8_t`) since neither of them can go above `255`.

Reply:
No, CPUs do not like 8 bit numbers very much. They are less efficient and generate larger machine code. Here the compiler will be smart enough to promote it to 32bit anyway, but in general we should not use uint8_t for counter variables or something like that. The counter will use exactly one CPU register anyway, thus there is no benefit in using a smaller data type.

Reply:
hmmm, thanks for clarifying